hw6/mergeSort/mergeSort.c

```diff
@@ -0,0 +1,201 @@
+﻿#define _CRT_SECURE_NO_WARNINGS
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "List.h"
+
+#define ARRAY_SIZE 40
+#define ERROR_STRING "err"
+
+void moveElements(List* listFrom, List* listTo, int counter)
+{
+    for (int i = 0; i < counter; ++i)
+    {
+        addElement(listTo, getHeadName(listFrom), getHeadPhone(listFrom));
+        deleteHeadElement(listFrom);
+    }
+}
+
+int compareHeads(List* firstList, List* secondList, int key)
+{
+    if (key == 1)
+    {
+        return (strcmp(getHeadName(firstList), getHeadName(secondList)) < 0) ? 1 : 2;
+    }
+    return (strcmp(getHeadPhone(firstList), getHeadPhone(secondList)) < 0) ? 1 : 2;
+}
+
+List* merge(List* firstList, List* secondList, int key)
+{
+    List* sortedList = createList();
+    while (!sizeOfList(firstList) == 0 && !sizeOfList(secondList) == 0)
+    {
+        compareHeads(firstList, secondList, key) == 1 ? moveElements(firstList, sortedList, 1) : moveElements(secondList, sortedList, 1);
```

Comment:
```suggestion
        compareHeads(firstList, secondList, key) == 1 
            ? moveElements(firstList, sortedList, 1) 
            : moveElements(secondList, sortedList, 1);
```
Меня пугают такие длинные строки и так много джействий на одной строке